### DISCUSSION TITLE
Upgraded installation

### DISCUSSION THREAD

**Discussion by JeanB51** [jgyates/genmon#1190] [Show and tell]
So I just completed an upgrade of my Genmon installation.

I had been running Genmon on a raspberry Pi Zero since 2017, and everything was running fine, so I never bothered to upgrade. Recently, the installation was getting flakier, and I had to restart regularly. I figured that it must be the SD card getting worn out. So I decided that now was the time for an upgrade.

Initially, I thought of replicating my existing setup. I bought a Pi Zero 2 for extra speed (the web page was indeed rather sluggish). I used a MAX 3232 board for TTL conversion to Rs232 (see first picture). After soldering everything, I had problems making it work. I figured it was probably due to my weak soldering skills, so I decided to change my approach – I would go fully modular.

![IMG_7487](https://github.com/user-attachments/assets/895b5662-05b1-43d5-8765-4bf55dd97b43)

I had a Pi3B available, so I ordered the Pintsize.me HAT with the RJ45 injector, as I have a long wire coming from the generator into the garage. The HAT worked perfectly. I did have some issues, with wiring, entirely of my own fault, but Jon Helms (@skipfire) was very responsive with ideas and comments to help me along. Following his suggestion, I have RX , TX and Gnd coming on the wire from the generator, and take power from a wall brick next to my project box.

See some pictures, I have the Molex connected into the Nexus controller, then converted to RJ45 with the injector. The wire runs into the garage, then into my project box, then is converted back into the Molex and into the HAT. I took advantage of the available pins on the Hat to power two LEDs on the box cover. I have an ON LED when power is applied, and a Ready light controlled by gengpio. 

I also am running Log2Ram to reduce wear on the SD card. I made sure that everything in the box is un-pluggable so I can replace any component in the future without any cutting or de-soldering of wires. Following from other projects here, I found great panel-mount connectors for RJ-45 and power.

![IMG_7485](https://github.com/user-attachments/assets/47115904-5c33-4e03-afd5-9cb302667fac)
![IMG_7570](https://github.com/user-attachments/assets/9ab379f9-adc9-4d17-b187-65f25912c9a4)
![IMG_7573](https://github.com/user-attachments/assets/3442dc9d-496c-4d59-beaa-ff576ff97233)
![IMG_7568](https://github.com/user-attachments/assets/471ca028-c0fb-444c-9e90-35882d22f108)

I installed Tailscale on the PI, and I can monitor the web page from remote sites easily and securely.

7 years later, I have to say, Genmon, already a great product back then, has really matured into an amazing tool. I am impressed with the addons page and the ease of turning features on and off. Also, the Pi3B is still a perfect platform for this, runing below 3% CPU.

Many thanks to JG Yates for the amazing work he has done, and the great ongoing support. 
Some beer money coming your way !.

Jean

**Comment by jgyates** [comment]
Hi @JeanB51 

Thanks for the kind feedback and the donation. Great setup pics!

**Reply by skipfire** [reply]
@JeanB51 agreed, thanks for sharing, even after making the hardware for over 3 years I still get a bit giddy every time I see the pics or videos about the installs when I can see the hardware.

**Comment by JeanB51** [comment]
Haha I understand.
Even me, every time I go into the garage, I sneak a peak into the box :-)
Having the 2 leds on the outside gives me great reassurance !
________________________________
De : Jon R. Helms ***@***.***>
Envoyé : mardi 1 octobre 2024 16:28
À : jgyates/genmon ***@***.***>
Cc : Jean Boisvert ***@***.***>; Mention ***@***.***>
Objet : Re: [jgyates/genmon] Upgraded installation (Discussion #1190)


@JeanB51<https://github.com/JeanB51> agreed, thanks for sharing, even after making the hardware for over 3 years I still get a bit giddy every time I see the pics or videos about the installs when I can see the hardware.
